Egardia: Adding configuration tag according to new docs

jeroenterheerdt opened the pull request:
A request was made in the last doc review to start using the %configuration% tag (PR: https://github.com/home-assistant/home-assistant.github.io/pull/4803). This PR does that. 
One thing is missing (need help!): I did not find a sample of how to specify the keys for the map type in configuration. The component accepts a list of states that can contain lists of codes, like this:
```yaml
report_server_codes:
  arm: XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX
  disarm: XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX
  armhome: XXXXXXXXXXXXXXXX
  triggered: XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX
  ignore: XXXXXXXXXXXXXXXX
```
not sure how to put this in the %configuration% tag. For now, I left both the old version and the {%configuration%} tag in.

**Description:**


**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

## Checklist:

- [X] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [X] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

jeroenterheerdt committed:
Egardia: Adding configuration tag according to new docs 

Egardia: Adding configuration tag according to new docs

jeroenterheerdt committed:
Egardia: Adding configuration tag according to new docs 

Egardia: Adding configuration tag according to new docs

jeroenterheerdt commented:
@balloob @MartinHjelmare here is the new replacement PR for the requested change to the Egardia docs. Not sure if I did it correctly though - for at least one piece I need some help (see description of PR). I closed #4876 since the rebase to `current` did not work. This new PR is based on `current`. Please help figuring out the right way of using the `%configuration%` tag for the list of lists that my component accepts.

bottomquark committed:
add more examples and clarify calendar.caldav

robmarkcole committed:
Fix typo

The extra () resulted in a 404 error

IainPHay committed:
Change upgrade homeassistant to upgrade home-assistant

Line 40 Change:
You can also use `hassbian-config` to automate the process by running `sudo hassbian-config upgrade homeassistant`
to
You can also use `hassbian-config` to automate the process by running `sudo hassbian-config upgrade home-assistant`
as this was incorrect

maxlaverse committed:
Improve caldav documentation

DubhAd committed:
Merge pull request #4898 from IainPHay/patch-2

Change upgrade homeassistant to upgrade home-assistant

DubhAd committed:
Merge pull request #4895 from robmarkcole/patch-11

Fix typo - extra ()

DubhAd committed:
Merge pull request #4899 from maxlaverse/improve_caldav_documentation

Improve caldav documentation

balloob committed:
Update virtualenv.markdown

bottomquark committed:
Merge branch 'current' into patch-1

balloob committed:
release 0.65.3

bottomquark committed:
fixes for the previous merge for #4896

removed duplicate example and duplicate text

balloob committed:
Remove broken redirect from (#4901)

balloob committed:
Update the used Ruby version

balloob committed:
Update footer to mention Netlify

emlove committed:
Configuration Flow Translations (#4880)

* translation_develop and config flow info

* Components must store strings in their own dir

balloob commented:
I don't know how to specify a list of tags either.

jeroenterheerdt commented:
@balloob if you don't know, who can? I mean, you are the master of this eco system 😀

MartinHjelmare commented:
Look at the mysensors hub docs. The `gateway` key holds a list of maps of key value pairs. You want a map of key value pairs where the values are lists. So just turn it around.

I don't think you should specify the strings as comma separated. Just make that regular lists of strings. The user can use both ways.

MartinHjelmare commented on source/_components/egardia.markdown:
````diff
@@ -59,9 +65,45 @@ You can change this, however, using the following procedure. This is a more adva
         triggered: XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX, XXXXXXXXXXXXXXXX
         ignore: XXXXXXXXXXXXXXXX
     ```
+{% configuration %}
````
Only one of these sections should be written and it should contain all the options.

jeroenterheerdt replied:
ok

jeroenterheerdt committed:
Updating egardia.markdown based on review.

jeroenterheerdt committed:
Updating egardia.markdown based on review.

jeroenterheerdt commented:
@MartinHjelmare not sure if I understand what you meant, anyway, here is a first stab at improving. Now only one `{% configuration %}` section remains and I added some headers with `{% linkable_title %}` tags. Not sure if the config is correct right now, because it seems the `mysensors` hub has a slightly different config.

MartinHjelmare commented on source/_components/egardia.markdown:
```diff
@@ -16,40 +17,90 @@ The `egardia` platform enables the ability to control an [Egardia](http://egardi
+  type: int
+  default: 52010
+report_server_codes:
+  description: List of codes for the different states.
```
This is a map of list of codes.

jeroenterheerdt replied:
ok

MartinHjelmare commented on source/_components/egardia.markdown:
```diff
@@ -16,40 +17,90 @@ The `egardia` platform enables the ability to control an [Egardia](http://egardi
+    arm:
+      description: List of codes for the 'arm' state.
+      required: false
+      type: map
```
I think you can specify `list` here, although it's not clearly mentioned in our docs for configuration tag.

jeroenterheerdt replied:
well, it is *not* mentioned, but let me try :)

jeroenterheerdt committed:
Update egardia.markdown

jeroenterheerdt committed:
Update egardia.markdown

jeroenterheerdt commented:
@MartinHjelmare build failed, not sure why.

MartinHjelmare commented:
Seems unrelated to this PR:
```
10:00:59 AM: failed during stage 'deploying site': Invalid filename 'getting-started/basic/#remote-access.html'. Deployed filenames cannot contain # or ? characters
```

DubhAd committed:
Added version note (#4900)

Since Synology offer an outdated Python, providing a note about how to install 0.64.x

DubhAd committed:
Bumped version to be explicitly 3.5.3 (#4888)

DubhAd committed:
Bumped version to be explicitly 3.5.3 (#4889)

DubhAd committed:
Bumped version to be explicitly 3.5.3 (#4890)

DubhAd committed:
Bumped to Python 3.6 from 3.4 (#4891)

Skipped 3.5 because it provides 3.5.1.

DubhAd committed:
Minimum Python version requirements (#4887)

DubhAd committed:
Provided note about python versions (#4892)

Synology Package Center provides only 3.5.1, nothing newer :(

DubhAd committed:
Added python version note (#4893)

DubhAd committed:
Bumped version to be explicitly 3.5.3 (#4886)

fabaff committed:
Update (fixes #4868) (#4905)

corneyl committed:
Added info on AND condition (#4867)

jeroenterheerdt commented:
so...?

MartinHjelmare commented:
I think @balloob fixed some of these build bugs in another PR, but I don't think this one was included. You could try to rebase your branch on current, though, to see if that helps.

DubhAd committed:
Merge pull request #4896 from bottomquark/patch-1

add more examples and clarify calendar.caldav

balloob committed:
release 0.65.4

balloob commented:
Yeah the build failed because your base branch is not up to date. Both `HEAD` of `current` and `next` branches are passing.

dshokouhi committed:
Add songpal to ignore list (#4906)

After updating 0.65.4 I kept seeing errors about a component I am not using or even have.  This change is to make sure the discovery docs are kept to upto date with devices added to the discovery component.

jeroenterheerdt committed:
Merge branch 'current' of https://github.com/jeroenterheerdt/home-assistant.github.io into current

jeroenterheerdt commented:
I tried rebasing, the result is above. Let me know if this is OK.

MartinHjelmare commented:
There's lots of unrelated commits in your branch now.

It's a lot harder to rebase if you're working locally directly on one of the main upstream branches, in this case `current`. Take for habit to always checkout and create a new, with another name, local branch based on one of the main upstream branches when starting your work. Then you can easily rebase the local "feature" branch during the development process

This is outlined here:
https://home-assistant.io/developers/development_submitting/
and here:
https://home-assistant.io/developers/development_catching_up/

jeroenterheerdt commented:
You are right, I forgot to create a new branch. So what do I do? Delete repo and refork?

jeroenterheerdt commented:
I tried creating a new fork, but the moment I create a new branch for my changes GitHub says my branch is X commits ahead and Y commits behind home-assistant.next, while I created my new branch based on `current`. Should I be creating my branch based on `next` instead?

jeroenterheerdt commented:
II created a new PR (replacement of the replacement) here: https://github.com/home-assistant/home-assistant.github.io/pull/4910. Hope this is correct this time.
